CMakeLists.txt

```diff
@@ -1,84 +1,72 @@
-cmake_minimum_required(VERSION 3.8)
-project(LittleTypeLibrary)
-
-if(MSVC)
```

Comment:
Did you forget to add this compiling arguments?

Reply:
I wasn't sure about adding these, since this command adds definitions globally, which is very much not desirable in the case where this project is imported without being installed to the system first.

Are these needed for the tests as build requirements? Are these usage requirements for dependents?

Reply:
I answered it in test/Cmakelist.txt if you want to read the answer :).